### ISSUE TITLE
Make WireMock extension for Quarkus an official Dev Service

### ISSUE THREAD

**Issue by oleg-nenashev** [quarkiverse/quarkus-wiremock#70]
Congrats with the 1.0.0 release @chberger !

This extension looks like a Dev Service, smells like a DevService, and also behaves and gets configured as a Dev Service. But formally it is not one at the moment. It would be great to make it more official and have it listed on https://quarkus.io/guides/dev-services and within the Quarkus app.

I'd be happy to help with documentation and whatever needed for it

## Proposal

- [x] Add extension to https://quarkus.io/guides/dev-services - https://github.com/quarkusio/quarkus/pull/39913
- [ ] Reference the extension tutorial from proper Dev Services locations, e.g. in https://quarkus.io/guides/getting-started-testing
- [x] Add the extension to the Dev Services list in Dev UI

## Quarkus Dev UI

Right now the  extension is not listed as a dev service

![image](https://github.com/quarkiverse/quarkus-wiremock/assets/3000480/f53f3444-cc3f-4eb5-a837-9eca36928baf)

![image](https://github.com/quarkiverse/quarkus-wiremock/assets/3000480/0e201a8f-a59a-4c46-8a41-8a907f48d5f3)

**Comment by chberger** [comment]
Hi @oleg-nenashev, many thanks! 🚀

I really like your proposal! 👍 However, I'm not sure whether the `Quarkus` team will actually list **external** Dev Services in their guides. I have the impression that all Dev Services currently mentioned are part of the core `platform`. 

Anyway, feel free to get in touch with the `Quarkus` team. Maybe it is actually conceivable ... 🤞

Regarding your last point I've created a PR which addresses this issue: https://github.com/quarkiverse/quarkus-wiremock/pull/72. I would be delighted if you would review it.

**Comment by oleg-nenashev** [comment]
Cc @maxandersen @aloubyansky for the core service. On behalf of the WireMock community, would be happy to cooperate on that

**Comment by maxandersen** [comment]
We already have neo4j listed. 

If gets to be issue we'll fix it and make it work.

**Comment by holly-cummins** [comment]
I've raised https://github.com/quarkusio/quarkusio.github.io/issues/1869 for a discussion of listing more dev services on https://quarkus.io/guides/dev-services.

**Comment by oleg-nenashev** [comment]
Hi @chberger could you please send me your contact email address (my one is public in the profile - `o.v.nenashev at gmail dot com`)? I would like to add you directly to the email conversation with @holly-cummins that we started on this subject

**Comment by oleg-nenashev** [comment]
@chberger could you please reopen it?

**Comment by chberger** [comment]
@oleg-nenashev thanks for the hint

**Comment by holly-cummins** [comment]
- I've raised https://github.com/quarkusio/quarkus/pull/39913 to list the WireMock extension in the list of dev services 
- For https://quarkus.io/guides/getting-started-testing, it looks like the page mentions WireMock a few times, and should be updated to refer to the extension instead. Perhaps @oleg-nenashev or @chberger could do a PR?
- For the Dev UI, this information is automagically pulled from what's in the extension. @phillip-kruger, is the dev services information supposed to be automatically populated if an extension has a dev service, or do the WireMock team need to work through some of the steps in https://quarkus.io/guides/dev-ui?

**Comment by holly-cummins** [comment]
Oh, and I thought I already commented, but I can't find it, so maybe not. As a general terminology clarification, a dev service is just the ability to start a zero-configuration container; some extensions do, and some don't. It's not a description of extension status or something extensions get nominated to be.

**Comment by phillip-kruger** [comment]
@holly-cummins yes that should happen automatically. Do you need details on how/why?

**Comment by holly-cummins** [comment]
Thanks, @phillip-kruger! I can see the dev services description build item [in the code](https://github.com/quarkiverse/quarkus-wiremock/blob/794ac69ceb603bb0fa037b80fc980f9ea6999ef0/deployment/src/main/java/io/quarkiverse/wiremock/devservice/WireMockServerProcessor.java#L81-L85), so I guess the next thing is to confirm if the dev service is still missing from the UI. 

I've just checked (doh!), and I can see it in my dev UI:

<img width="1584" alt="image" src="https://github.com/quarkiverse/quarkus-wiremock/assets/11509290/cead1ea4-7f21-4ad7-ae6b-b1adea00ad9c">

@oleg-nenashev, is it still missing for you?

**Comment by oleg-nenashev** [comment]
@holly-cummins it was added by @chberger in 1.1.0 - https://github.com/quarkiverse/quarkus-wiremock/pull/72 . Sorry, this ticket is a bit dated, will update

**Comment by holly-cummins** [comment]
See https://quarkus.io/version/main/guides/dev-services for the new guide content. So I think it's just mentioning the extension in https://quarkus.io/guides/getting-started-testing which is outstanding.

**Comment by oleg-nenashev** [comment]
Not stale but still needs a few steps

**Comment by holly-cummins** [comment]
The one outstanding item for this item is adding something to https://quarkus.io/guides/getting-started-testing. I never know exactly how much the guides should be about the core/platform, and how much they should include the whole Quarkiverse, but independently of that question, this bit in the guide seems definitely wrong:

> Common cases include starting docker containers using [Testcontainers](https://www.testcontainers.org/) (an example of which can be found [here](https://github.com/quarkusio/quarkus/blob/main/test-framework/keycloak-server/src/main/java/io/quarkus/test/keycloak/server/KeycloakTestResourceLifecycleManager.java)), or starting a mock HTTP server using [Wiremock](https://wiremock.org/) (an example of which can be found [here](https://github.com/geoand/quarkus-test-demo/blob/main/src/test/java/org/acme/getting/started/country/WiremockCountries.java)).

We wouldn't expect people to use a test resource for WireMock, because there's an extension. So the text wants changing, maybe to find a different example of user-managed resources. And @geoand, I wonder if the [code example](https://github.com/geoand/quarkus-test-demo/blob/main/src/test/java/org/acme/getting/started/country/WiremockCountries.java) should be updated?

**Comment by geoand** [comment]
Yeah, we can definitely update that

**Comment by wjglerum** [comment]
I proposed to add a little more information for the Wiremock dev ui card here https://github.com/quarkiverse/quarkus-wiremock/pull/180

If you have any other thoughts or additions let me know 😄

**Comment by oleg-nenashev** [comment]
Not stale, but close to be completed
